src/Api/Serializer/AbstractSerializer.php

```diff
@@ -83,6 +94,16 @@ public function getAttributes($model, array $fields = null)
 
         $attributes = $this->getDefaultAttributes($model);
 
+        if (isset(static::$attributeHandlers[static::class])) {
+            foreach (static::$attributeHandlers[static::class] as $handler) {
```

Comment:
Ah I just noticed I need to check parent classes as well and a test case for that.